File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/models/WPFTransactionTypes.java
package com.emerchantpay.gateway.genesisandroid.api.models;

import java.util.Date;

public class WPFTransactionTypes {

    private String name;
    private String bin;
    private String tail;

    // Default variable
    private Boolean def;

    private Date expirationDate;

    //WPF Transaction types
    public static String authorize = "authorize";
    public static String authorize3d = "authorize3d";
    public static String sale = "sale";
    public static String sale3d = "sale3d";
    public static String initRecurringSale = "init_recurring_sale";
    public static String initRecurringSale3d = "init_recurring_sale3d";
    public static String ezeewallet = "ezeewallet";
    public static String sofort = "sofort";
    public static String cashu = "cashu";
    public static String paysafecard = "paysafecard";
    public static String ppro = "ppro";
    public static String neteller = "neteller";
    public static String poli = "poli";
    public static String p24 = "p24";
    public static String citadelPayin = "citadel_payin";
    public static String idebitPayin = "idebit_payin";
    public static String instaDebitPayin = "insta_debit_payin";
    public static String paypalExpress = "paypal_express";
    public static String webmoney = "webmoney";
    public static String sddSale = "sdd_sale";
    public static String sddInitRecurringSale = "sdd_init_recurring_sale";
    public static String trustlySale = "trustly_sale";
    public static String trustlyWithdrawal = "trustly_withdrawal";
    public static String wechat = "wechat";
    public static String rpnPayment = "rpn_payment";
    public static String paysec = "paysec";
    public static String paysecPayout = "paysec_payout";
    public static String intersolve = "intersolve";
    public static String fashioncheque = "fashioncheque";
    public static String containerStore = "container_store";
    public static String neosurf = "neosurf";
    public static String klarnaAuthorize = "klarna_authorize";

    public WPFTransactionTypes() {

    }

    public WPFTransactionTypes(String name) {
        super();

        this.name = name;
    }

    public String getTransactionName() {
        return name;
    }
}

File: GenesisAndroid/src/main/java/com/emerchantpay/gateway/genesisandroid/api/models/WPFTransactionTypes.kt
package com.emerchantpay.gateway.genesisandroid.api.models

import java.util.*

class WPFTransactionTypes {

    private var transactionName: String = ""
    private val bin: String? = null
    private val tail: String? = null

    // Default const valiable
    private val def: Boolean? = null

    private val expirationDate: Date? = null

    constructor() {

    }

    constructor(name: String) : super() {
        this.transactionName = name
    }

    companion object {

        //WPF Transaction types
        const val authorize = "authorize"
        const val authorize3d = "authorize3d"
        const val sale = "sale"
        const val sale3d = "sale3d"
        const val initRecurringSale = "init_recurring_sale"
        const val initRecurringSale3d = "init_recurring_sale3d"
        const val ezeewallet = "ezeewallet"
        const val sofort = "sofort"
        const val cashu = "cashu"
        const val paysafecard = "paysafecard"
        const val ppro = "ppro"
        const val neteller = "neteller"
        const val poli = "poli"
        const val p24 = "p24"
        const val citadelPayin = "citadel_payin"
        const val idebitPayin = "idebit_payin"
        const val instaDebitPayin = "insta_debit_payin"
        const val paypalExpress = "paypal_express"
        const val webmoney = "webmoney"
        const val sddSale = "sdd_sale"
        const val sddInitRecurringSale = "sdd_init_recurring_sale"
        const val trustlySale = "trustly_sale"
        const val trustlyWithdrawal = "trustly_withdrawal"
        const val wechat = "wechat"
        const val rpnPayment = "rpn_payment"
        const val paysec = "paysec"
        const val paysecPayout = "paysec_payout"
        const val intersolve = "intersolve"
        const val fashioncheque = "fashioncheque"
        const val containerStore = "container_store"
        const val neosurf = "neosurf"
        const val klarnaAuthorize = "klarna_authorize"
    }
}
